onnxruntime/core/framework/utils.cc

```diff
@@ -160,14 +168,31 @@ const std::string& GetNodeInputProviderType(const SessionState::NodeInfo& info)
   return required_provider_type;
 }
 
+template <class T, class Batch>
+Status BatchOrCopyHelper(const SessionState& session_state,
```

Comment:
>BatchOrCopyHelper [](http://example.com/codeflow?start=7&length=17)

it's only used by a static function - not sure if function templates can also be declared static but it could be in `namespace { }`